File: apps/ios/Core/TextLimits.swift
import Foundation

/// Shared caps for learner-authored text. Both clients must refuse or stop accepting
/// input past these lengths instead of silently truncating on save/send.
public enum TextLimits {
    public static let typedReplyCharacters = 2_000
    public static let correctionCharacters = 10_000

    public static func clampTypedReply(_ text: String) -> String {
        String(text.prefix(typedReplyCharacters))
    }

    public static func clampCorrection(_ text: String) -> String {
        String(text.prefix(correctionCharacters))
    }

    public static func typedReplyExceedsLimit(_ text: String) -> Bool {
        text.count > typedReplyCharacters
    }

    public static func correctionExceedsLimit(_ text: String) -> Bool {
        text.count > correctionCharacters
    }
}

File: apps/android/app/src/main/java/chat/mural/core/TextLimits.kt
package chat.mural.core

/** Shared caps for learner-authored text. Both clients must refuse or stop accepting
 *  input past these lengths instead of silently truncating on save/send. */
object TextLimits {
    const val TYPED_REPLY_CHARACTERS = 2_000
    const val CORRECTION_CHARACTERS = 10_000

    fun clampTypedReply(text: String): String = text.take(TYPED_REPLY_CHARACTERS)
    fun clampCorrection(text: String): String = text.take(CORRECTION_CHARACTERS)
    fun typedReplyExceedsLimit(text: String): Boolean = text.length > TYPED_REPLY_CHARACTERS
    fun correctionExceedsLimit(text: String): Boolean = text.length > CORRECTION_CHARACTERS
}
